src/controller/AutoCommissioner.cpp

```diff
@@ -481,8 +537,26 @@ CHIP_ERROR AutoCommissioner::CommissioningStepFinished(CHIP_ERROR err, Commissio
         }
         else if (report.Is<NetworkCommissioningStatusInfo>())
         {
+            // This report type is used when an error happens in either NetworkConfig or ConnectNetwork commands
             completionStatus.networkCommissioningStatus =
                 MakeOptional(report.Get<NetworkCommissioningStatusInfo>().networkCommissioningStatus);
+
+            // if we are configured to scan networks, then don't error out
+            // instead, allow the app to try another network
+            if ((mParams.GetAttemptWiFiNetworkScan().ValueOr(false) &&
+                 mDeviceCommissioningInfo.network.wifi.endpoint != kInvalidEndpointId) ||
+                (mParams.GetAttemptThreadNetworkScan().ValueOr(false) &&
+                 mDeviceCommissioningInfo.network.thread.endpoint != kInvalidEndpointId))
+            {
+                if (completionStatus.err == CHIP_NO_ERROR)
+                {
+                    completionStatus.err = err;
+                }
+                err = CHIP_NO_ERROR;
+                // walk back the completed stage to kScanNetworks
+                // this will allow the app to try another network
```

Comment:
```suggestion
                // Walk back the completed stage to kScanNetworks.
                // This will allow the app to try another network.
```

Reply:
fixed